temporaltest/server_test.go

```diff
@@ -0,0 +1,412 @@
+		temporaltest.WithT(t),
+		temporaltest.WithBaseWorkerOptions(
+			worker.Options{
+				MaxConcurrentActivityExecutionSize:      1,
+				MaxConcurrentLocalActivityExecutionSize: 1,
+			},
+		),
+	)
+
+	ts.NewWorker("hello_world", func(registry worker.Registry) {
+		RegisterWorkflowsAndActivities(registry)
+	})
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	wfr, err := ts.GetDefaultClient().ExecuteWorkflow(
+		ctx,
+		client.StartWorkflowOptions{TaskQueue: "hello_world"},
+		Greet,
+		"world",
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var result string
+	if err := wfr.Get(ctx, &result); err != nil {
+		t.Fatal(err)
+	}
+
+	if result != "Hello world" {
+		t.Fatalf("unexpected result: %q", result)
+	}
+}
+
+type denyAllClaimMapper struct{}
+
+func (denyAllClaimMapper) GetClaims(*authorization.AuthInfo) (*authorization.Claims, error) {
+	// Return claims that have no permissions within the cluster.
+	return &authorization.Claims{
+		Subject:    "test-identity",
+		System:     authorization.RoleUndefined,
+		Namespaces: nil,
+		Extensions: nil,
+	}, nil
+}
+
+func TestBaseServerOptions(t *testing.T) {
+	// This test verifies that we can set custom claim mappers and authorizers
+	// with BaseServerOptions.
+	ts := temporaltest.NewServer(
+		temporaltest.WithT(t),
+		temporaltest.WithBaseServerOptions(
+			temporal.WithClaimMapper(func(cfg *config.Config) authorization.ClaimMapper {
+				return denyAllClaimMapper{}
+			}),
+			temporal.WithAuthorizer(authorization.NewDefaultAuthorizer()),
+		),
+	)
+
+	_, err := ts.GetDefaultClient().ExecuteWorkflow(
+		context.Background(),
+		client.StartWorkflowOptions{},
+		"test-workflow",
+	)
+	if err == nil {
+		t.Fatal("err must be non-nil")
+	}
+
+	if !strings.Contains(err.Error(), authorization.RequestUnauthorized) {
+		t.Errorf("expected error %q, got %q", authorization.RequestUnauthorized, err)
+	}
+}
+
+func TestClientWithCustomInterceptor(t *testing.T) {
+	var opts client.Options
+	opts.Interceptors = append(opts.Interceptors, NewTestInterceptor())
+	ts := temporaltest.NewServer(
+		temporaltest.WithT(t),
+		temporaltest.WithBaseClientOptions(opts),
+	)
+
+	ts.NewWorker(
+		"hello_world",
+		func(registry worker.Registry) {
+			RegisterWorkflowsAndActivities(registry)
+		},
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	wfr, err := ts.GetDefaultClient().ExecuteWorkflow(
+		ctx,
+		client.StartWorkflowOptions{TaskQueue: "hello_world"},
+		Greet,
+		"world",
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var result string
+	if err := wfr.Get(ctx, &result); err != nil {
+		t.Fatal(err)
+	}
+
+	if result != "Hello world" {
+		t.Fatalf("unexpected result: %q", result)
+	}
+}
+
+func TestSearchAttributeCacheDisabled(t *testing.T) {
+	// TODO(jlegrone) re-enable this test when advanced visibility is enabled in temporalite.
+	t.Skip("This test case does not currently pass as of the 1.20 release")
```

Comment:
try enabling this again now?

Reply:
I've updated the test. I think there may have been a regression in the 1.20 release that still exists which I've left a TODO for. But the test now verifies that we can register a custom search attribute and list workflows which set it.